lib/truffle/truffle/patching.rb

```diff
@@ -29,7 +40,7 @@ class Module
   private :autoload_without_truffle_patching
 
   def autoload(const, path)
-    if Truffle::Patching::TRUFFLE_PATCHES[path]
+    if Truffle::Patching::TRUFFLE_AFTER_PATCHES[path] || Truffle::Patching::TRUFFLE_BEFORE_PATCHES[path]
       require path
```

Comment:
@eregon This is the issue ... the current code is expecting OpenSSL to autoload:
Here is the autoload: https://github.com/graalvm/truffleruby/blob/master/lib/mri/net/http.rb#L27

That's required when I do `gem install`, etc by rubygems and bundler too maybe